### ISSUE TITLE
Build broke on Alpine Linux

### ISSUE THREAD

**Issue by 1ma** [crazyxman/simdjson_php#6]
from https://github.com/crazyxman/simdjson_php/issues/5#issuecomment-522953414

I cannot build the latest commit under Alpine Linux 3.10, getting these errors:

```
$ make
/bin/sh /root/simdjson_php/libtool --mode=compile cc  -I. -I/root/simdjson_php -DPHP_ATOM_INC -I/root/simdjson_php/include -I/root/simdjson_php/main -I/root/simdjson_php -I/usr/include/php7 -I/usr/include/php7/main -I/usr/include/php7/TSRM -I/usr/include/php7/Zend -I/usr/include/php7/ext -I/usr/include/php7/ext/date/lib  -DHAVE_CONFIG_H  -g -O2   -c /root/simdjson_php/simdjson.c -o simdjson.lo 
mkdir .libs
 cc -I. -I/root/simdjson_php -DPHP_ATOM_INC -I/root/simdjson_php/include -I/root/simdjson_php/main -I/root/simdjson_php -I/usr/include/php7 -I/usr/include/php7/main -I/usr/include/php7/TSRM -I/usr/include/php7/Zend -I/usr/include/php7/ext -I/usr/include/php7/ext/date/lib -DHAVE_CONFIG_H -g -O2 -c /root/simdjson_php/simdjson.c  -fPIC -DPIC -o .libs/simdjson.o
In file included from /root/simdjson_php/simdjson.c:23:
/root/simdjson_php/src/bindings.h:25:1: error: unknown type name 'namespace'; did you mean 'isspace'?
 namespace simdjsonphp {
 ^~~~~~~~~
 isspace
/root/simdjson_php/src/bindings.h:25:23: error: expected '=', ',', ';', 'asm' or '__attribute__' before '{' token
 namespace simdjsonphp {
                       ^
/root/simdjson_php/src/bindings.h:41:1: error: unknown type name 'bool'; did you mean '_Bool'?
 bool cplus_simdjson_isvalid(const char *json);
 ^~~~
 _Bool
/root/simdjson_php/simdjson.c:56:22: error: conflicting types for 'cplus_simdjson_isvalid'
 extern unsigned char cplus_simdjson_isvalid(const char *json);
                      ^~~~~~~~~~~~~~~~~~~~~~
In file included from /root/simdjson_php/simdjson.c:23:
/root/simdjson_php/src/bindings.h:41:6: note: previous declaration of 'cplus_simdjson_isvalid' was here
 bool cplus_simdjson_isvalid(const char *json);
      ^~~~~~~~~~~~~~~~~~~~~~
make: *** [Makefile:193: simdjson.lo] Error 1
```

Commit `a0e942d` from May 29th builds fine.

If you want to try to reproduce, I started from a clean `alpine:3.10` Docker container and only needed to install these packages: `apk add git build-base php7-dev`

**Comment by lemire** [comment]
I get the same issue under Ubuntu:

```
$ make && make install
/bin/bash /home/dlemire/CVS/github/simdjson_php/libtool --mode=compile gcc-8  -I. -I/home/dlemire/CVS/github/simdjson_php -DPHP_ATOM_INC -I/home/dlemire/CVS/github/simdjson_php/include -I/home/dlemire/CVS/github/simdjson_php/main -I/home/dlemire/CVS/github/simdjson_php -I/usr/include/php/20151012 -I/usr/include/php/20151012/main -I/usr/include/php/20151012/TSRM -I/usr/include/php/20151012/Zend -I/usr/include/php/20151012/ext -I/usr/include/php/20151012/ext/date/lib  -DHAVE_CONFIG_H  -g -O2   -c /home/dlemire/CVS/github/simdjson_php/simdjson.c -o simdjson.lo
libtool: compile:  gcc-8 -I. -I/home/dlemire/CVS/github/simdjson_php -DPHP_ATOM_INC -I/home/dlemire/CVS/github/simdjson_php/include -I/home/dlemire/CVS/github/simdjson_php/main -I/home/dlemire/CVS/github/simdjson_php -I/usr/include/php/20151012 -I/usr/include/php/20151012/main -I/usr/include/php/20151012/TSRM -I/usr/include/php/20151012/Zend -I/usr/include/php/20151012/ext -I/usr/include/php/20151012/ext/date/lib -DHAVE_CONFIG_H -g -O2 -c /home/dlemire/CVS/github/simdjson_php/simdjson.c  -fPIC -DPIC -o .libs/simdjson.o
In file included from /home/dlemire/CVS/github/simdjson_php/simdjson.c:23:
/home/dlemire/CVS/github/simdjson_php/src/bindings.h:25:1: error: unknown type name ‘namespace’; did you mean ‘isspace’?
 namespace simdjsonphp {
 ^~~~~~~~~
 isspace
/home/dlemire/CVS/github/simdjson_php/src/bindings.h:25:23: error: expected ‘=’, ‘,’, ‘;’, ‘asm’ or ‘__attribute__’ before ‘{’ token
 namespace simdjsonphp {
                       ^
/home/dlemire/CVS/github/simdjson_php/src/bindings.h:41:1: error: unknown type name ‘bool’; did you mean ‘_Bool’?
 bool cplus_simdjson_isvalid(const char *json);
 ^~~~
 _Bool
/home/dlemire/CVS/github/simdjson_php/simdjson.c:56:22: error: conflicting types for ‘cplus_simdjson_isvalid’
 extern unsigned char cplus_simdjson_isvalid(const char *json);
                      ^~~~~~~~~~~~~~~~~~~~~~
In file included from /home/dlemire/CVS/github/simdjson_php/simdjson.c:23:
/home/dlemire/CVS/github/simdjson_php/src/bindings.h:41:6: note: previous declaration of ‘cplus_simdjson_isvalid’ was here
 bool cplus_simdjson_isvalid(const char *json);
      ^~~~~~~~~~~~~~~~~~~~~~
Makefile:193: recipe for target 'simdjson.lo' failed
make: *** [simdjson.lo] Error 1
```

**Comment by crazyxman** [comment]
Thank you for your feedback, the next update will fix these issues

**Comment by c-locksmith** [comment]
I would really love to try this in my current project, since it heavily relies on loading many json files, but unfortunately this error is still present it seems.

 I am on centos7 and get the same error.
`$ make && make install
/bin/sh /home/vagrant/simdjson_php/libtool --mode=compile cc  -I. -I/home/vagrant/simdjson_php -DPHP_ATOM_INC -I/home/vagrant/simdjson_php/include -I/home/vagrant/simdjson_php/main -I/home/vagrant/simdjson_php -I/usr/include/php -I/usr/include/php/main -I/usr/include/php/TSRM -I/usr/include/php/Zend -I/usr/include/php/ext -I/usr/include/php/ext/date/lib  -DHAVE_CONFIG_H  -g -O2   -c /home/vagrant/simdjson_php/simdjson.c -o simdjson.lo
libtool: compile:  cc -I. -I/home/vagrant/simdjson_php -DPHP_ATOM_INC -I/home/vagrant/simdjson_php/include -I/home/vagrant/simdjson_php/main -I/home/vagrant/simdjson_php -I/usr/include/php -I/usr/include/php/main -I/usr/include/php/TSRM -I/usr/include/php/Zend -I/usr/include/php/ext -I/usr/include/php/ext/date/lib -DHAVE_CONFIG_H -g -O2 -c /home/vagrant/simdjson_php/simdjson.c  -fPIC -DPIC -o .libs/simdjson.o
In file included from /home/vagrant/simdjson_php/simdjson.c:23:0:
/home/vagrant/simdjson_php/src/bindings.h:25:1: error: unknown type name 'namespace'
 namespace simdjsonphp {
 ^
/home/vagrant/simdjson_php/src/bindings.h:25:23: error: expected '=', ',', ';', 'asm' or '__attribute__' before '{' token
 namespace simdjsonphp {
                       ^
/home/vagrant/simdjson_php/src/bindings.h:41:1: error: unknown type name 'bool'
 bool cplus_simdjson_isvalid(const char *json);
 ^
/home/vagrant/simdjson_php/simdjson.c:56:22: error: conflicting types for 'cplus_simdjson_isvalid'
 extern unsigned char cplus_simdjson_isvalid(const char *json);
                      ^
In file included from /home/vagrant/simdjson_php/simdjson.c:23:0:
/home/vagrant/simdjson_php/src/bindings.h:41:6: note: previous declaration of 'cplus_simdjson_isvalid' was here
 bool cplus_simdjson_isvalid(const char *json);
      ^
make: *** [simdjson.lo] Error 1
`
Switching back to the commit that was suggested above did not work for me.

**Comment by lemire** [comment]
The error looks like a C++-to-C issue. The library itself is in C++. Surely, PHP can support extensions written in C++? Somehow?

**Comment by sandrokeil** [comment]
@crazyxman Any news on this?

**Comment by crazyxman** [comment]
@sandrokeil  Great,  You are welcome to fixed it.

**Comment by lemire** [comment]
@WeaponMan : would you issue a pull request ? It seems like @crazyxman  would take it.

**Comment by WeaponMan** [comment]
@lemire @sandrokeil is already on it, so it will be double work

**Comment by sandrokeil** [comment]
It's included in #10 and this is superseded by #11
